zipline/_protocol.pyx

```diff
@@ -391,8 +399,74 @@ cdef class BarData:
             normalize_date(self.simulation_dt_func())
         )
 
+    #################
+    # OLD API SUPPORT
+    #################
+    cdef _calculate_universe(self):
```

Comment:
This could be memoized using `remember_last` with the addition of `dt` as parameter.